NCS.DSS.Interaction/PatchInteractionHttpTrigger/Function/PatchInteractionHttpTrigger.cs

```diff
@@ -45,78 +45,124 @@ public PatchInteractionHttpTrigger(IResourceHelper resourceHelper, IHttpRequestH
                 return new BadRequestObjectResult(HttpStatusCode.BadRequest);
             }
 
             var ApimURL = _httpRequestMessageHelper.GetDssApimUrl(req);
             if (string.IsNullOrEmpty(ApimURL))
             {
-                log.LogInformation("Unable to locate 'apimurl' in request header");
+                _logger.LogInformation("Unable to locate 'apimurl' in request header");
                 return new BadRequestObjectResult(HttpStatusCode.BadRequest);
             }
 
-            log.LogInformation("Patch Interaction C# HTTP trigger function processed a request. " + touchpointId);
+            _logger.LogInformation("Patch Interaction C# HTTP trigger function processed a request. " + touchpointId);
 
             if (!Guid.TryParse(customerId, out var customerGuid))
+            {
+                _logger.LogWarning("Unable to parse 'customerId' to a GUID. Customer GUID: {CustomerID}", customerId);
                 return new BadRequestObjectResult(customerGuid);
+            }
 
             if (!Guid.TryParse(interactionId, out var interactionGuid))
+            {
+                _logger.LogWarning("Unable to parse 'interactionId' to a GUID. Interaction ID: {InteractionId}", interactionId);
                 return new BadRequestObjectResult(interactionGuid);
+            }
 
+            _logger.LogInformation("Attempting to retrieve resource from request.");
             InteractionPatch interactionPatchRequest;
-
             try
             {
                 interactionPatchRequest = await _httpRequestMessageHelper.GetResourceFromRequest<Models.InteractionPatch>(req);
             }
             catch (Exception ex)
             {
+                _logger.LogError(ex, "Unable to parse {interactionPatchRequest} from request body. Exception: {ExceptionMessage}", nameof(interactionPatchRequest), ex.Message);
                 return new UnprocessableEntityObjectResult(_dynamicHelper.ExcludeProperty(ex, ["TargetSite"]));
             }
 
             if (interactionPatchRequest == null)
+            {
+                _logger.LogWarning("{interactionPatchRequest} object is NULL.", nameof(interactionPatchRequest));
                 return new UnprocessableEntityObjectResult(req);
+            }
 
             interactionPatchRequest.LastModifiedTouchpointId = touchpointId;
 
+            _logger.LogInformation("Attempting to validate {interactionPatchRequest} object", nameof(interactionPatchRequest));
             var errors = _validate.ValidateResource(interactionPatchRequest);
 
             if (errors != null && errors.Any())
+            {
+                _logger.LogWarning("Falied to validate {outcomeValidationObject} object", nameof(interactionPatchRequest));
                 return new UnprocessableEntityObjectResult(errors);
+            }
+            _logger.LogInformation("Successfully validated {outcomeValidationObject} object", nameof(interactionPatchRequest));
+
```

Comment:
Need a log stating that input validation is complete, see https://github.com/SkillsFundingAgency/dss-transfer/pull/31/files#diff-695bbf80412b14a3c3dcfde0ff6fd17ff9d7119b08dcd1741520ff6e7ee8cd9f line 71 for an example